http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -703,26 +727,39 @@ public void onContent(Response response, ByteBuffer content)
         // process response
         long responseStart = System.nanoTime();
 
-        JettyResponse jettyResponse = null;
         T value;
-        try {
-            jettyResponse = new JettyResponse(response, listener.getInputStream());
-            value = responseHandler.handle(request, jettyResponse);
+
+        if (isStreaming) {
+            JettyResponse jettyResponse;
+            try {
+                jettyResponse = new JettyResponse(response, listener.getInputStream());
```

Comment:
Can we extract this above the `if` statement ? Do we need to wrap between `try`

Reply:
I don't think that would help. In the non-streaming case it gets the entire value after allocating the `JettyResponse`. The streaming case doesn't do that.